blockstack/lib/nameset/virtualchain_hooks.py

```diff
@@ -585,9 +585,10 @@ def db_save( block_height, consensus_hash, ops_hash, accepted_ops, virtualchain_
                 # check if the required number of blocks have been mined since threshold crossed
                 if (block_height - threshold_block_id == v2_upgrade_signal_import_threshold):
                     # emit parsable log entry and export datafile
-                    log.warn('[v2-upgrade] import threshold reached at block: {}'.format(block_height))
-                    db_state.set_v2_import_block_reached(block_height)
-                    db_state.perform_v2_upgrade_datafile_export(consensus_hash)
+                    export_block_id = db_state.get_current_block()
+                    log.warn('[v2-upgrade] import threshold reached at block: {}'.format(export_block_id))
+                    db_state.set_v2_import_block_reached(export_block_id)
+                    db_state.perform_v2_upgrade_datafile_export(export_block_id)
```

Comment:
Do you know how long this method will run in production?  Because if it takes more than 10 minutes (a likely possibility, given how many rows there are to process), then all nodes with this patch can fall behind the Bitcoin chain tip.

Reply:
This discussion should probably be in https://github.com/blockstack/stacks-blockchain/pull/1910 where the logic was implemented?

Reply:
Note: running this on my local machine against mainnet originally took 40+ minutes. @wileyj said it takes a similar amount of time on our prod infrastructure. 

I've added a few optimizations that brings the time down to less than 10 minutes on my machine.

Reply:
Good good